docs/setup-mac.md

```diff
@@ -97,3 +97,26 @@ See [./setup-gmail.md](./setup-gmail.md) for instructions on setting up Gmail to
     ```sh
     yarn workspace server run db:seed
     ```
+
+## Datadog
```

Comment:
I'm not sure that basic setup steps for working on this repo should direct folks to use Datadog at all –it should rarely be necessary for local environments, and I'd like to think that's the exception (reserved for advanced debugging/investigation scenarios) rather than the norm. I think there are a few reasons why:
1. Cost. Although I don't think it's a problem for people to generate and send local runtime/APM metrics to Datadog for specific cases, I think it _should_ be reserved for intentional investigation, rather than something that's "on by default". Otherwise, we'll end up paying for a lot of throw-away development data that nobody ever uses. 
2. Related to cost, I think having these instructions geared towards a one-time setup as a system service will inspire "set-and-forget" behavior, and I would like to make sure that as much as possible, folks turn off local Datadog reporting once a specific debugging exercise or investigative effort is concluded. 
3. Development complexity, and a desire to avoid having contributors go through additional setup steps for their dev environment when they might not actually be using Datadog (since my guess is most people won't).

To turn all this into a concrete ask for this PR:
1. It's worth splitting out this Datadog section into a separate, standalone document, rather than bundling it into dev environment setup instructions.
2. Within that separate document, add some introductory text (maybe around the content that's currently on line 103) that clarifies when you might actually want/need to do this. In particular, I think it should be a generally driven by a team consensus ("yes, this is something we need to look into, and there are reasons why it would be useful for being able to experiment in local development").